File: 13-roman-to-integer/roman-to-integer.cpp
class Solution {
public:
    int romanToInt(string s) {
        unordered_map<char, int> umap;
        umap['I'] = 1;
        umap['V'] = 5;
        umap['X'] = 10;
        umap['L'] = 50;
        umap['C'] = 100;
        umap['D'] = 500;
        umap['M'] = 1000;
        int n = s.length();
        int result = 0;
        for(int i=0;i<n-1;i++)
        {
            if(umap[s[i+1]] > umap[s[i]])
            {
                result-=umap[s[i]];
            }
            else
            {
                result+=umap[s[i]];
            }
        }
        result+=umap[s[n-1]];

        return result;
        
    }
};

File: 13-roman-to-integer/roman-to-integer.java
class Solution {
    public int romanToInt(String s) {
        int number = 0;

        HashMap<Character,Integer> umap = new HashMap<>();

        umap.put('I', 1);
        umap.put('V', 5);
        umap.put('X', 10);
        umap.put('L', 50);
        umap.put('C', 100);
        umap.put('D', 500);
        umap.put('M', 1000);

        int n = s.length();
        for(int i=0;i<s.length()-1;i++)
        {
            char ch = s.charAt(i);
            int digit = umap.get(ch);

            char ch2 = s.charAt(i+1);
            if(umap.get(ch2) > digit)
            {
                number-= digit;
            }
            else
                number+= digit;
        }

        number+= umap.get(s.charAt(n-1));
        return number;
        
    }
}